File: 알고리즘/Math/O_bj2745.cpp
/*
n진법을 10진법으로 바꾸는 문제

풀이 1
1. 알파벳이 들어오면 char - char를 이용해 숫자로 변환한 뒤 10을 더한다.
2. 각 자리수에는 (진법^자리수)를 곱해준다.
*/

#include <iostream>
#include <string>

using namespace std;

int main(void){

    string x;
    int n;
    cin >> x >> n;

    int result = 0;
    int mul = 1;

    for(int i = x.size()-1; i >= 0; i--){
        int num;
        if('A' <= x[i] && x[i] <= 'Z'){
            num = (int)(x[i]-'A') + 10;
        }
        else{
            num = (int)(x[i]-'0');
        }
        result += num*mul;
        mul *= n;
    }
    printf("%d\n",result);
}

File: 알고리즘/Math/O_bj2745.kt
import java.util.*

class bj2745 {
}

/**
 * B진법 수 N이 주어진다.
 * 이 수를 10진법으로 바꿔라
 * */

fun main(){
    val br = System.`in`.bufferedReader()
    val st = StringTokenizer(br.readLine())

    val n = st.nextToken()
    val b = st.nextToken().toInt()

    var result = 0
    var mul = 1
    for(i in n.length-1 downTo 0){
        result += charToInt(n[i])*mul
        mul *= b
    }

    println(result)
}

fun charToInt(c: Char): Int{
    if(c in 'A'..'Z') return 10 + c.code-'A'.code
    return c.code-'0'.code
}